pyxem/utils/dask_tools.py

```diff
@@ -200,6 +256,42 @@ def _process_dask_array(
     return output_array
 
 
+def _get_iter_array(iter_array, dask_array):
+    if len(iter_array.shape) > len(dask_array.shape):
```

Comment:
Could we get a one long docstring for this function (I realise it's private, but it would make life/debugging attempts easier)

Reply:
Agreed. I went a bit overboard and wrote a very extensive docstring here. As this functionality require some very specific inputs, and if it is even slightly wrong, you can get some really strange results which are difficult to decipher.